modules/jsdom/src/polyfills/glfw.ts

```diff
@@ -0,0 +1,425 @@
+    ({xscale: window._xscale, yscale: window._yscale} = glfw.getWindowContentScale(id));
+
+    !window._forceNewWindow && !rootWindow && (rootWindow = window);
+    window._frameBufferWidth  = window._width * window._xscale;
+    window._frameBufferHeight = window._height * window._yscale;
+    window._subscriptions && window._subscriptions.unsubscribe();
+    window._subscriptions = new Subscription();
+    window.cursor         = window._cursor;
+
+    glfw.swapInterval(window.swapInterval);
+    glfw.swapBuffers(id);
+
+    [dndEvents(window).subscribe(onGLFWDndEvent.bind(window)),
+     mouseEvents(window).subscribe(onGLFWMouseEvent.bind(window)),
+     wheelEvents(window).subscribe(onGLFWWheelEvent.bind(window)),
+     windowEvents(window).subscribe(onGLFWWindowEvent.bind(window)),
+     keyboardEvents(window).subscribe(onGLFWKeyboardEvent.bind(window)),
+    ].forEach((subscription) => window._subscriptions.add(subscription));
+  } catch (e) {
+    console.error('Error creating window:', e);
+    window.destroyGLFWWindow();
+    throw e;
+  }
+
+  return window;
+}
+
+function onGLFWDndEvent(this: DOMWindow, event: GLFWDndEvent) {
+  dispatchGLFWEvent(this, event, this.Event);
+}
+
+function onGLFWMouseEvent(this: DOMWindow, event: GLFWMouseEvent) {
+  this.mouseX   = event.x;
+  this._mouseY  = event.y;
+  this._buttons = event.buttons;
+  let m         = this._modifiers;
+  m             = event.altKey ? (m | GLFWModifierKey.MOD_ALT) : (m & ~GLFWModifierKey.MOD_ALT);
+  m = event.ctrlKey ? (m | GLFWModifierKey.MOD_CONTROL) : (m & ~GLFWModifierKey.MOD_CONTROL);
+  m = event.metaKey ? (m | GLFWModifierKey.MOD_SUPER) : (m & ~GLFWModifierKey.MOD_SUPER);
+  m = event.shiftKey ? (m | GLFWModifierKey.MOD_SHIFT) : (m & ~GLFWModifierKey.MOD_SHIFT);
+  m = event.capsLock ? (m | GLFWModifierKey.MOD_CAPS_LOCK) : (m & ~GLFWModifierKey.MOD_CAPS_LOCK);
+  this._modifiers = m;
+  dispatchGLFWEvent(this, event, this.MouseEvent);
+}
+
+function onGLFWWheelEvent(this: DOMWindow, event: GLFWWheelEvent) {
+  this._scrollX += event.deltaX;
+  this._scrollY += event.deltaY;
+  dispatchGLFWEvent(this, event, this.WheelEvent);
+}
+
+function onGLFWWindowEvent(this: DOMWindow, event: GLFWWindowEvent) {
+  this._x                 = event.x;
+  this._y                 = event.y;
+  this._width             = event.width;
+  this._height            = event.height;
+  this._xscale            = event.xscale;
+  this._yscale            = event.yscale;
+  this._focused           = event.focused;
+  this._minimized         = event.minimized;
+  this._maximized         = event.maximized;
+  this._frameBufferWidth  = event.frameBufferWidth;
+  this._frameBufferHeight = event.frameBufferHeight;
+  this._devicePixelRatio =
+    Math.min(this._frameBufferWidth / this._width, this._frameBufferHeight / this._height);
+  dispatchGLFWEvent(this, event, this.Event);
+  if (event.type === 'close') { this._destroyGLFWWindow(); }
```

Comment:
```suggestion
  if (event.type === 'close') { this_destroyGLFWWindow(); }
```

Reply:
You corrected this to 
`this_destroyGLFWWindow()`
did you mean
`this.destroyGLFWWindow()`
?

Reply:
Yep I typo'd the suggestion. Thought the method was still private with an underscore.